### DISCUSSION TITLE
What code language it can be?

### DISCUSSION THREAD

**Discussion by kedybogan** [community/community#57952] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Hi, is this CSS so web based software for that Kiosk or instead of web based software, is it something like Java and JavaFX?
https://www.youtube.com/watch?v=Oquf3pPtkeY
Feel free to tell anything.

**Comment by airtower-luna** [comment, marked as answer]
Could be either, a browser running in kiosk mode with a web application, or a native application running in fullscreen mode (and some native GUI frameworks also use CSS!). There is no fundamental reason it needs to be one or the other, either can be good or bad depending on how well it's made. :wink:

**Reply by kedybogan** [reply]
Thank you so much for the clarification! Very helpful to me :). I will go through web-based software but someone also told me "You can use html css in .exe" and "search PWA" but the end of the day I will definitely go through CSS etc. front end stuff. JavaFX doesn't sound good to me.
I was concerned because transitions were instant in the YouTube video like checkout etc. so no web page load/request waiting that's why I thought it could be a "native application" that's what we call it I guess.
But yeah now I can say that I'm sure about web-based software will work.

**Reply by airtower-luna** [reply]
> But yeah now I can say that I'm sure about web-based software will work.

It *can* work, you still need to get it right. :wink: 

Depending on what your kiosk is for expect to run the web application on a *local* web server. For example if you're building an ordering system for a restaurant you wouldn't want it to go down if the internet connection of that restaurant has issues, and you might need to access local devices (e.g. for payment).
